datawig/simple_imputer.py

```diff
@@ -130,21 +131,199 @@ def check_data_types(self, data_frame: pd.DataFrame) -> None:
         logger.info("Assuming {} string input columns: {}".format(len(self.string_columns),
                                                                   ", ".join(self.string_columns)))
 
+    def fit_hpo_new(self,
```

Comment:
This method looks very heavy. Consider splitting in a small methods where you add comments to explain what's going on. 

I'd rather see something like this or similar 

```
def method():
  set_defaults()
  validated_inputs()
  for conf in configurations:
    try_conf() 
```

Reply:
Yes, very good point.